components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/designgenerator/generators/EdgesGenerator.java

```diff
@@ -533,4 +539,15 @@ private NodeType getSiddhiElementType(SiddhiElementConfig siddhiElementConfig) t
     private static String generateEdgeId(String parentID, String childID) {
         return String.format("%s_%s", parentID, childID);
     }
+
+    private SiddhiElementConfig getElementWithSinkCorrelateId(String correlateId) {
+        for (SourceSinkConfig sourceConfig : siddhiAppConfig.getSourceList()) {
+            if (sourceConfig.getCorrelateId() != null) {
+                if (sourceConfig.getCorrelateId().equals(correlateId)) {
```

Comment:
Do you need two levels of if condition for this? We can merge this together

Reply:
Fixed